Source/Urho3D/Physics/PhysicsWorld.cpp

```diff
@@ -402,7 +402,11 @@ void PhysicsWorld::Raycast(ea::vector<PhysicsRaycastResult>& result, const Ray&
     for (int i = 0; i < rayCallback.m_collisionObjects.size(); ++i)
     {
         PhysicsRaycastResult newResult;
-        newResult.body_ = static_cast<RigidBody*>(rayCallback.m_collisionObjects[i]->getUserPointer());
+        Component *component = static_cast<Component *>(rayCallback.m_collisionObjects[i]->getUserPointer());
```

Comment:
I'm not 100% sure about that, looks kludge-y. Lemme think...

Reply:
yep. alternatively - you need a complete physics representation. or maybe replace dynamic casts with component->getnode->getgigidbody / component->getnode->getkinematiccharacter